### PR TITLE
Add tests for webhook selector

### PR DESCRIPTION
**Pull request by howardjohn**
This changes no behavior, just adding tests

### CODE REVIEW

**Comment by howardjohn** [comment]
cc @Monkeyanator @zerobfd this will impact the new labeling stuff

File: operator/cmd/mesh/manifest-generate_test.go [hunk 1]
```diff
@@ -703,3 +708,281 @@ func generateManifest(inFile, flags string, chartSource chartSourceType) (string
...
+		// Auto injection
+		{
+			name:     "auto",
+			webhooks: autoWebhook,
+			checks: []test{
+				{empty, empty, "istiod"},
+				{canaryRevLabel, empty, ""},
+				{legacyLabel, empty, "istiod"},
+				{legacyLabelDisabled, empty, ""},
+				// TODO: This is sort of a weird behavior. We should still be able to explicitly select the revision
```
Comment (richardwxn) [inline operator/cmd/mesh/manifest-generate_test.go:881]: is there a doc that documents all of these behaviors, specifically combination with auto injection case, for some cases it it not really intuitive

Comment (howardjohn) [reply operator/cmd/mesh/manifest-generate_test.go:881]: No, it's undocumented alpha experimental behavior. I am just trying to capture the current state so we can improve on it

**Review by richardwxn — APPROVED** [review]

**Pull request merged**